js/symbol/shaping.js

```diff
@@ -38,38 +38,35 @@ function Shaping(positionedGlyphs, text, top, bottom, left, right, writingMode)
 
 const newLine = 0x0a;
 
-function shapeText(text, glyphs, maxWidth, lineHeight, horizontalAlign, verticalAlign, justify, spacing, translate, verticalHeight, writingMode) {
+function breakLines(text, lineBreakPoints) {
+    const lines = [];
+    let start = 0;
+    lineBreakPoints.forEach(lineBreak => {
+        lines.push(text.substring(start, lineBreak));
+        start = lineBreak;
+    });
```

Comment:
Please use a `for (const lineBreak of lineBreakPoints) { ... }` loop here instead of `forEach`.